File: python-tests/OOP/employee.py
class Employee:
    __emp_count = 0 # class variable

    def __init__(self,name="",salary=0.0): # instance method
        self.__name = name  # instance variables, note that they are set to private with double "__" at the start
        self.__salary = salary
        Employee.__emp_count += 1

    def set_name(self,name):
        self.__name = name

    def get_name(self): # encapsulation of data members
        return self.__name

    def set_salary(self,salary):
        self.__salary = salary

    def get_salary(self):
        return self.__salary

    def __str__(self):
        return "Name : " + self.__name + " Salary : " + str(self.__salary)

    def get_employee_count(): # class method
        return Employee.__emp_count

File: cpp-tests/employee.cpp
#include <iostream>
#include <string>

using namespace std;

class Employee 
{
private:
string name;
float salary;

public:
Employee(string name, float salary)
{
name = name;
salary = salary;
}

void set_name(string name)
{
name = name;
}

string get_name()
{
cout<<name;
return *name;
}

void set_salary(float salary)
{
cout<<salary;
salary = salary;
}

float get_salary()
{
return *salary;
}

};

int main()
{

Employee e("Julie",3839.938);
e.set_name("Julie");
e.set_salary(7362.93);
cout<<e.get_name();
cout<<e.get_salary();

}